src/gazebo_mavlink_interface.cpp

```diff
@@ -1117,38 +1124,41 @@ void GazeboMavlinkInterface::handle_actuator_controls() {
   // Read Input References
   input_reference_.resize(n_out_max);
 
+  // Get the normalized (-1 ... +1) actuator controls input from PX4 instance
   Eigen::VectorXd actuator_controls = mavlink_interface_->GetActuatorControls();
+
   if (actuator_controls.size() < n_out_max) return; //TODO: Handle this properly
+
+  // Translate actuator control input into motor input reference [rad/s]
   for (int i = 0; i < input_reference_.size(); i++) {
     if (armed) {
       input_reference_[i] = (actuator_controls[input_index_[i]] + input_offset_[i])
           * input_scaling_[i] + zero_position_armed_[i];
-      // std::cout << input_reference_ << ", ";
+
     } else {
       input_reference_[i] = zero_position_disarmed_[i];
-      // std::cout << input_reference_ << ", ";
+
     }
   }
   // std::cout << "Input Reference: " << input_reference_.transpose() << std::endl;
   received_first_actuator_ = mavlink_interface_->GetReceivedFirstActuator();
 }
 
+// set joint positions
 void GazeboMavlinkInterface::handle_control(double _dt)
 {
-  // set joint positions
   for (int i = 0; i < input_reference_.size(); i++) {
     if (joints_[i] || joint_control_type_[i] == "position_gztopic") {
       double target = input_reference_[i];
-      if (joint_control_type_[i] == "velocity")
-      {
+
+      // Apply PID based force for velocity & PID controlled joint
+      if (joint_control_type_[i] == "velocity" && joint_uses_pid_[i]) {
```

Comment:
what happens if `joint_uses_pid_[i]=false`? Wouldn't it fall into https://github.com/PX4/PX4-SITL_gazebo/pull/877/files#diff-b6e0a79395aaa68135ca1c7ed7c931289bc2f154042330e4f3a7a00ed99ffba2R1200 ?